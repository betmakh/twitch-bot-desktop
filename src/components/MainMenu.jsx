import React from 'react';
import { MenuItem, MenuList } from 'material-ui/Menu';
import Paper from 'material-ui/Paper';
import List, { ListItem, ListItemIcon, ListItemText } from 'material-ui/List';
import SettingsIcon from 'material-ui-icons/Settings';
import ChatIcon from 'material-ui-icons/Chat';
import Grid from 'material-ui/Grid';
import CastIcon from 'material-ui-icons/Cast';
import Drawer from 'material-ui/Drawer';
import Select from 'material-ui/Select';
import ViewListIcon from 'material-ui-icons/ViewList';
import { withStyles } from 'material-ui/styles';
import Divider from 'material-ui/Divider';
import Avatar from 'material-ui/Avatar';

import {
	CHAT_COMPONENT,
	USER_LIST_COMPONENT,
	SETTINGS_COMPONENT,
	FOLLOWERS_LIST_COMPONENT
} from '../utils/constants.js';

const styles = theme => ({
	itemActive: {
		background: theme.palette.common.faintBlack
	},
	logo: {
		marginRight: theme.spacing.unit,
		maxWidth: theme.spacing.unit * 5
	},
	drawerHeader: theme.mixins.toolbar
});

class MainMenu extends React.Component {
	handleSectionSelect(sectionName) {
		this.props.selectSection(sectionName);
	}

	render() {
		const { classes, sectionSelected, channels, currentChannel, saveSettings, channelData } = this.props;
		// if (!currentChannel && channels) {
		// 	saveSettings({ currentChannel: channels[0] });
		// }
		return (
			<Drawer type="permanent">
				<MenuItem className={classes.drawerHeader}>
					{channelData && <Avatar alt="Remy Sharp" src={channelData.logo} className={classes.logo} />}
					{channels && channels.length ? (
						<Select
							value={currentChannel.toString()}
							fullWidth
							onChange={event => {
								saveSettings({ currentChannel: event.target.value });
							}}
							className={classes.selectEmpty}
						>
							{channels.map(channel => (
								<MenuItem key={channel} value={channel}>
									{channel}
								</MenuItem>
							))}
						</Select>
					) : (
						'No channels configured'
					)}
				</MenuItem>
				<Divider />
				<MenuList style={{ width: this.props.drawerWidth }}>
					<MenuItem
						onClick={this.handleSectionSelect.bind(this, CHAT_COMPONENT)}
						selected={sectionSelected === CHAT_COMPONENT}
					>
						<ListItemIcon>
							<ChatIcon />
						</ListItemIcon>Chat
					</MenuItem>
					<MenuItem
						onClick={this.handleSectionSelect.bind(this, USER_LIST_COMPONENT)}
						selected={sectionSelected === USER_LIST_COMPONENT}
					>
						<ListItemIcon>
							<ViewListIcon />
						</ListItemIcon>List of watchers
					</MenuItem>
					<MenuItem
						onClick={this.handleSectionSelect.bind(this, FOLLOWERS_LIST_COMPONENT)}
						selected={sectionSelected === FOLLOWERS_LIST_COMPONENT}
					>
						<ListItemIcon>
							<CastIcon />
						</ListItemIcon>Followers list
					</MenuItem>
					<MenuItem
						onClick={this.handleSectionSelect.bind(this, SETTINGS_COMPONENT)}
						selected={sectionSelected === SETTINGS_COMPONENT}
					>
						<ListItemIcon>
							<SettingsIcon />
						</ListItemIcon>Setting
					</MenuItem>
				</MenuList>
			</Drawer>
		);
	}
}

export default withStyles(styles)(MainMenu);
