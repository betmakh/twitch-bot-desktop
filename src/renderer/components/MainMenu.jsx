import React from "react";
import { MenuItem, MenuList } from "@material-ui/core/Menu";
import Paper from "@material-ui/core/Paper";
import List, {
  ListItem,
  ListItemIcon,
  ListItemText,
} from "@material-ui/core/List";
import SettingsIcon from "@material-ui/icons/Settings";
import ChatIcon from "@material-ui/icons/Chat";
import Grid from "@material-ui/core/Grid";
import CastIcon from "@material-ui/icons/Cast";
import Drawer from "@material-ui/core/Drawer";
import Select from "@material-ui/core/Select";
import ViewListIcon from "@material-ui/icons/ViewList";
import { withStyles } from "@material-ui/core/styles";
import Divider from "@material-ui/core/Divider";
import Avatar from "@material-ui/core/Avatar";

import {
  CHAT_COMPONENT,
  USER_LIST_COMPONENT,
  SETTINGS_COMPONENT,
  FOLLOWERS_LIST_COMPONENT,
} from "../utils/constants.js";

const styles = (theme) => ({
  itemActive: {
    background: theme.palette.common.faintBlack,
  },
  logo: {
    marginRight: theme.spacing.unit,
    maxWidth: theme.spacing.unit * 5,
  },
  drawerHeader: theme.mixins.toolbar,
});

class MainMenu extends React.Component {
  handleSectionSelect(sectionName) {
    this.props.saveSettings({ sectionSelected: sectionName });
    this.props.selectSection(sectionName);
  }

  render() {
    const {
      classes,
      sectionSelected,
      channels,
      currentChannel,
      saveSettings,
      channelData,
      widgetUrl,
      drawerWidth,
    } = this.props;
    // if (!currentChannel && channels) {
    // 	saveSettings({ currentChannel: channels[0] });
    // }
    return (
      <Drawer variant="permanent">
        <MenuItem className={classes.drawerHeader}>
          {channelData && (
            <Avatar
              alt="Remy Sharp"
              src={channelData.logo}
              className={classes.logo}
            />
          )}
          {channels && channels.length ? (
            <Select
              value={currentChannel.toString()}
              fullWidth
              onChange={(event) => {
                saveSettings({
                  currentChannel: event.target.value,
                });
              }}
              className={classes.selectEmpty}
            >
              {channels.map((channel) => (
                <MenuItem key={channel} value={channel}>
                  {channel}
                </MenuItem>
              ))}
            </Select>
          ) : (
            "No channels configured"
          )}
        </MenuItem>
        <Divider />
        <MenuList style={{ width: this.props.drawerWidth }}>
          <MenuItem
            onClick={this.handleSectionSelect.bind(this, CHAT_COMPONENT)}
            selected={sectionSelected === CHAT_COMPONENT}
          >
            <ListItemIcon>
              <ChatIcon />
            </ListItemIcon>
            Chat
          </MenuItem>
          <MenuItem
            onClick={this.handleSectionSelect.bind(this, USER_LIST_COMPONENT)}
            selected={sectionSelected === USER_LIST_COMPONENT}
          >
            <ListItemIcon>
              <ViewListIcon />
            </ListItemIcon>
            List of watchers
          </MenuItem>
          <MenuItem
            disabled
            onClick={this.handleSectionSelect.bind(
              this,
              FOLLOWERS_LIST_COMPONENT
            )}
            selected={sectionSelected === FOLLOWERS_LIST_COMPONENT}
          >
            <ListItemIcon>
              <CastIcon />
            </ListItemIcon>
            Followers list
          </MenuItem>
          <MenuItem
            onClick={this.handleSectionSelect.bind(this, SETTINGS_COMPONENT)}
            selected={sectionSelected === SETTINGS_COMPONENT}
          >
            <ListItemIcon>
              <SettingsIcon />
            </ListItemIcon>
            Setting
          </MenuItem>
        </MenuList>
        {widgetUrl && (
          <webview
            src={widgetUrl}
            style={{
              display: "inline-flex",
              width: `${drawerWidth}px`,
              maxWidth: "100%",
              overflow: "hidden",
              height: "480px",
            }}
          />
        )}
      </Drawer>
    );
  }
}

export default withStyles(styles)(MainMenu);
