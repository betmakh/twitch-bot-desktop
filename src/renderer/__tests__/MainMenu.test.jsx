import React from 'react';
// import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';

import MainMenu from '../components/MainMenu';
import EnzymeAdapter from '../utils/EnzymeConfig';

test('Main menu renders correctly', () => {
	const component = shallow(<MainMenu />);
	// const component = renderer.create(<MainMenu />);
	// let tree = component.toJSON();
	expect(component).toMatchSnapshot();
});
