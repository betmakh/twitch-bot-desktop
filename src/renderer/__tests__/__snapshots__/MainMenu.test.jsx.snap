// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main menu renders correctly 1`] = `
<MainMenu
  classes={
    Object {
      "drawerHeader": "MainMenu-drawerHeader-3",
      "itemActive": "MainMenu-itemActive-1",
      "logo": "MainMenu-logo-2",
    }
  }
/>
`;
